GenumKit/Parsers/AssetsCatalogParser.swift

```diff
@@ -8,24 +8,42 @@ import Foundation
 import PathKit
 
 public final class AssetsCatalogParser {
-  var imageNames = [String]()
+  var entries = [Entry]()
 
   public init() {}
 
   @discardableResult
   public func addImage(named name: String) -> Bool {
-    if imageNames.contains(name) {
+    if (entries.contains { $0.name == name }) {
       return false
     } else {
-      imageNames.append(name)
+      entries.append(Entry(name: name, value: name))
       return true
     }
   }
 
   public func parseCatalog(at path: String) {
     guard let items = loadAssetCatalog(at: path) else { return }
+
     // process recursively
-    processCatalog(items: items)
+    entries = process(items: items)
+  }
+
+  struct Entry {
+    var name: String
+    var value: String
+    var items: [Entry]?
```

Comment:
I think the first solution (a simple enum with associated values) would be easier to use than a struct with enum inside.

Using a `struct` with inner `enum` is only useful if all cases of the enum share multiple properties, like enough to want to avoid repeating each property in each enum case's associated values.
But here we only have one common property so I think it's not worth it and integrating `name` directly in each enum case would be easier to manipulate.

And that way you won't need dedicate inits, you can just create `Entry.image(name: "foo", value: "parent/foo")` and `Entry.namespace(name: "parent", entries: children)`.

PS: Note that I'm not sure if the `indirect` keyword is really needed. I think so, because the case references the enum being defined, but it's worth trying without to be sure we can't avoid it.